app/src/main/java/org/simple/clinic/onboarding/OnboardingScreen.kt

```diff
@@ -2,157 +2,251 @@ package org.simple.clinic.onboarding
+                viewModel.dispatch(event = GetStartedClicked)
+              }
+          )
+        }
+      }
+    }
   }
 
   override fun openOnboardingConsentScreen() {
     router.clearHistoryAndPush(OnboardingConsentScreenFragment.Key())
   }
 
-  private fun setIntroOneTextView() {
-    val onboardingEmphasisTextSpan = TextAppearanceSpan(context, R.style.TextAppearance_Simple_Subtitle1_Medium)
-    val onboardingEmphasisColorSpan = ForegroundColorSpan(requireContext().resolveColor(attrRes = R.attr.colorOnBackground))
-
-    val introOneFormattedString = buildSpannedString {
-      append(requireContext().getString(R.string.screenonboarding_intro_1))
-
-      inSpans(onboardingEmphasisTextSpan, onboardingEmphasisColorSpan) {
-        append(requireContext().getString(R.string.screenonboarding_intro_1_hypertension))
-      }
-    }
-
-    introOneTextView.text = introOneFormattedString
+  @Parcelize
+  data class Key(
+      override val analyticsName: String = "Onboarding Screen"
+  ) : ScreenKey() {
+    override fun instantiateFragment() = OnboardingScreen()
   }
 
-  private fun setIntroTwoTextView() {
-    val introTwoFormattedString = buildSpannedString {
-      append(requireContext().getString(R.string.screenonboarding_intro_2))
-
-      inSpans(
-          TextAppearanceSpan(context, R.style.TextAppearance_Simple_Subtitle1_Medium),
-          ForegroundColorSpan(requireContext().resolveColor(attrRes = R.attr.colorOnBackground))
-      ) {
-        append(requireContext().getString(R.string.screenonboarding_intro_2_bp))
-      }
-
-      append(requireContext().getString(R.string.screenonboarding_intro_2_and))
+  interface Injector {
+    fun inject(target: OnboardingScreen)
+  }
+}
 
-      inSpans(
-          TextAppearanceSpan(context, R.style.TextAppearance_Simple_Subtitle1_Medium),
-          ForegroundColorSpan(requireContext().resolveColor(attrRes = R.attr.colorOnBackground))
-      ) {
-        append(requireContext().getString(R.string.screenonboarding_intro_2_medicines))
-      }
+@Composable
+private fun OnboardingScreen(
+    onGetStartedClick: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+  Scaffold(
+      modifier = Modifier
+          .fillMaxSize()
+          .then(modifier),
+      bottomBar = {
+        FilledButtonWithFrame(
+            onClick = onGetStartedClick,
+        ) {
+          Text(
+              text = stringResource(R.string.onboarding_get_started).uppercase()
+          )
+        }
+      },
+      backgroundColor = SimpleTheme.colors.material.surface,
+  ) { innerPadding ->
+    val emphasisSpanStyle = SpanStyle(
+        color = SimpleTheme.colors.material.onBackground,
+        fontSize = SimpleTheme.typography.subtitle1Medium.fontSize,
+        fontStyle = SimpleTheme.typography.subtitle1Medium.fontStyle,
+        fontFamily = SimpleTheme.typography.subtitle1Medium.fontFamily,
+        fontWeight = SimpleTheme.typography.subtitle1Medium.fontWeight,
+    )
```

Comment:
Can we instead do SimpleTheme.typography.subtitle1Medium.toSpanStyle() ?

Reply:
Didn't knew this existed, that's why had to separate out to individual functions. This will resolve the rest of the issues. Thanks